### PR TITLE
Stabilize [T]::rotate_{left,right}

### PR DESCRIPTION
**Pull request by frewsxcv**
https://github.com/rust-lang/rust/issues/41891

### CODE REVIEW

File: src/liballoc/slice.rs [hunk 1]
```diff
@@ -1470,23 +1468,15 @@ impl<T> [T] {
...
     ///
     /// ```
-    /// #![feature(slice_rotate)]
-    ///
     /// let mut a = ['a', 'b', 'c', 'd', 'e', 'f'];
     /// a[1..5].rotate_left(1);
     /// assert_eq!(a, ['a', 'c', 'd', 'e', 'b', 'f']);
     /// ```
-    #[unstable(feature = "slice_rotate", issue = "41891")]
+    #[stable(feature = "slice_rotate", issue = "41891")]
```
Comment (scottmcm) [inline src/liballoc/slice.rs]: I think this should be `since = "1.26.0"`, not `issue`?

Comment (frewsxcv) [reply src/liballoc/slice.rs]: yep! good catch

**Review by scottmcm — APPROVED** [review]

**Comment by alexcrichton** [comment]
Despite having only been added very recently (the `rotate_{left,right}` methods) we've had a method `rotate` for quite some time now. I think that the left/right-taking name seems much better and would be comfortable stabilizing this even though these *specific* apis only landed recently. Curious what others think tough!

@rfcbot fcp merge

**Comment by sfackler** [comment]
Yep, I'm happy with the left/right rename.

**Pull request merged**